astropy/io/votable/tree.py

```diff
@@ -4285,8 +4285,8 @@ def to_xml(
         }
         kwargs.update(self._get_version_checks())
 
-        with util.convert_to_writable_filelike(fd, compressed=compressed) as fd:
-            w = XMLWriter(fd)
+        with util.convert_to_writable_filelike(fd, compressed=compressed) as fh:
```

Comment:
I am really not sure but does `with something(fd, ...) as fd` allow the old `fd` to be garbage collected faster?

Reply:
I would imagine that `fh` should hold a strong reference on `fd`, so my guess is no.